backend/src/packages/business/business.service.ts

```diff
@@ -105,6 +119,42 @@ class BusinessService implements IService {
 
     return await this.businessRepository.delete(id);
   }
+
+  public async createDriver({
+    payload,
+    id,
+  }: DriverCreatePayload): Promise<DriverAddResponseWithGroup> {
+    const doesBusinessExist = await this.findById(id);
+
+    if (!doesBusinessExist) {
+      throw new HttpError({
+        status: HttpCode.BAD_REQUEST,
+        message: HttpMessage.BUSINESS_DOES_NOT_EXIST,
+      });
+    }
+
+    return await this.driverService.create({ payload, id });
```

Comment:
I agree with all the async/await deletions except this one. Because here we don't just pass the data on to driverService.create, but before that we make a request for the existence of such a business to the business service method findById, which already forces us to use an asynchronous method.